### PR TITLE
[CDF-20939] Fix deploy order.

### PR DESCRIPTION
**Pull request by doctrino**
# Description

Please describe the change you have made.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/cdf_tk/load/_base_loaders.py [hunk 1]
```diff
@@ -255,6 +255,14 @@ def deploy_resources(
         self.build_path = path
         filepaths = self.find_files(path)
 
+        def sort_key(p: Path) -> int:
+            if result := re.findall(r"^(\d+)", p.stem):
+                return int(result[0])
+            else:
+                return len(filepaths)
+
+        filepaths = sorted(filepaths, key=sort_key)
```
Comment (gregertw) [inline cognite_toolkit/cdf_tk/load/_base_loaders.py:264]: Would be good with a comment on what you are trying to achieve here. It is not immediately obvious why you need a custom sort.

File: cognite_toolkit/cdf_tk/templates/_templates.py [hunk 2]
```diff
@@ -359,8 +359,16 @@ def process_config_files(
...
+
         # Sort to support 1., 2. etc prefixes
-        filepaths.sort()
+        def sort_key(p: Path) -> int:
+            if result := re.findall(r"^(\d+)", p.stem):
+                return int(result[0])
+            else:
+                return len(filepaths)
+
+        filepaths = sorted(filepaths, key=sort_key)
```
Comment (gregertw) [inline cognite_toolkit/cdf_tk/templates/_templates.py:370]: Same here

File: tests/tests_unit/approval_client.py [hunk 3]
```diff
@@ -555,23 +555,44 @@ def return_value(*args, **kwargs):
         method = available_post_methods[mock_method]
         return method
 
-    def dump(self) -> dict[str, Any]:
+    def dump(self, sort: bool = True) -> dict[str, Any]:
         """This returns a dictionary with all the resources that have been created and deleted.
 
+        The sorting is useful in snapshot testing, as it makes for a consistent output. If you want to check the order
```
Comment (gregertw) [inline tests/tests_unit/approval_client.py:561]: When we do API calls, the order of the calls is important. Don't you remove the ability to test that calls are executed in the correct order?

**Review by gregertw — COMMENTED** [review]
See comments

File: tests/tests_unit/approval_client.py (hunk 3 above)
Comment (doctrino) [reply tests/tests_unit/approval_client.py:561]: No, I enable it with this PR. 

For snapshot testing you want to sort all resources as you do not care about the order the calls are made, but to test the deploy order I needed to enable checking the order of the files.

**Review by ronpal — APPROVED** [review]

**Pull request merged**